pkg/tagger/global.go

```diff
@@ -57,6 +57,15 @@ func Tag(entity string, cardinality collectors.TagCardinality) ([]string, error)
 	return defaultTagger.Tag(entity, cardinality)
 }
 
+// TagWithHash is similar to Tag but it also computes and returns the hash of the tags found
+func TagWithHash(entity string, cardinality collectors.TagCardinality) ([]string, string, error) {
+	tags, err := Tag(entity, cardinality)
+	if err != nil {
+		return tags, "", err
+	}
+	return tags, computeTagsHash(tags), nil
```

Comment:
`computeTagsHash` here is not equivalent to `GetEntityHash`: the former computes the hash on the cardinality requested, while `GetEntityHash` will always get the hash from all tags regardless of the requested cardinality. I suppose that some consumers might not care, but having it on the tagger itself might cause surprising behavior. WDYT?

Reply:
well apparently AD was the only consumer of `GetEntityHash` - I'm thinking we have to remove `GetEntityHash` from the tagger in this PR, wdyt?

Reply:
`GetEntityHash` is racy by design if you want to identify the tags of a given entity. I believe `TagWithHash` should be the standard way to get consistent tags hash for futur consumers

Reply:
Agree on the racy design, so if we can get rid of it in the same PR, sounds good to me 👍

Reply:
cool thanks 👍  will do

Reply:
PR updated, didn't remove `GetEntityHash` but updated its logic, in some parts in AD we're interested in getting just the hash